lib/Dancer2/CLI/Gen.pm

```diff
@@ -0,0 +1,322 @@
+    format     => 's',
+    format_doc => 'directory',
+    required   => 0,
+    default    => sub{
+        my $self = shift; 
+        catdir( $self->parent_command->_dist_dir, 'skel' ); 
+    },
+);
+
+# Last chance to validate args before we attempt to do something with them
+sub BUILD {
+    my ( $self, $args ) = @_;
+
+    $self->osprey_usage( 1, qq{ 
+Invalid application name. Application names must not contain single colons, 
+dots, hyphens or start with a number.
+    }) unless is_module_name( $self->application );
+
+    my $path = $self->path;
+    -d $path or $self->osprey_usage( 1, "path: directory '$path' does not exist" );
+    -w $path or $self->osprey_usage( 1, "path: directory '$path' is not writeable" );
+
+    if ( my $skel = $self->skel ) {
+        -d $skel or $self->osprey_usage( 1, "skel: directory '$skel' not found" );
+    }
+}
+
+sub run {
+    my $self = shift;
+
+    my $app_name = $self->application;
+    my $app_file = $self->_get_app_file( $app_name );
+    my $app_path = $self->_get_app_path( $self->path, $app_name );
+
+    if( my $dir = $self->directory ) {
+        $app_path = catdir( $self->path, $dir );
+    }
+
+    my $files_to_copy = $self->_build_file_list( $self->skel, $app_path );
+    foreach my $pair( @$files_to_copy ) {
+        if( $pair->[0] =~ m/lib\/AppFile.pm$/ ) {
+            $pair->[1] = catfile( $app_path, $app_file );
+            last;
+        }
+    }
+
+    my $vars = {
+        appname          => $app_name,
+        appfile          => $app_file,
+        appdir           => File::Spec->rel2abs( $app_path ),
+        perl_interpreter => $self->_get_perl_interpreter,
+        cleanfiles       => $self->_get_dashed_name( $app_name ),
+        dancer_version   => $self->parent_command->_dancer2_version,
+    };
+
+    $self->_copy_templates( $files_to_copy, $vars, $self->overwrite );
+    $self->_create_manifest( $files_to_copy, $app_path );
+    $self->_add_to_manifest_skip( $app_path);
+
+    $self->_check_yaml;
+    $self->_how_to_run( $app_path );
+}
+
+sub _check_yaml {
+    if ( ! eval { require_module( 'YAML' ); 1; } ) {
+        print qq{
+*****
+
+WARNING: YAML.pm is not installed.  This is not a full dependency, but is highly
+recommended; in particular, the scaffolded Dancer app being created will not be
+able to read settings from the config file without YAML.pm being installed.
+
+To resolve this, simply install YAML from CPAN, for instance using one of the
+following commands:
+
+  cpan YAML
+  perl -MCPAN -e 'install YAML'
+  curl -L https://cpanmin.us | perl - --sudo YAML
+
+*****
+        };
+    }
+}
+
+sub _how_to_run {
+    my( $self, $app_path ) = @_;
+    print qq{
+Your new application is ready! To run it:
+
+        cd $app_path
+        plackup bin/app.psgi
+
+If you need community assistance, the following resources are available:
+- Dancer website: http://perldancer.org
+- Mailing list: http://lists.preshweb.co.uk/mailman/listinfo/dancer-users
```

Comment:
Probably looks more professional on the project to use lists.perldancer.org there (it's a configured vhost, it'll work) - and also decouples it from me in case hosting of the list ever needs to be moved elsewhere - bus factor and all that :)

Reply:
LOL, duly noted. Changed!